chronos_bin/src/message_receiver.rs

```diff
@@ -18,70 +20,111 @@ pub struct MessageReceiver {
 }
 
 impl MessageReceiver {
-    // pub fn new(consumer: Arc<KafkaConsumer>, producer: Arc<KafkaProducer>, data_store: Arc<Pg>) -> Self {
-    //     Self {
-    //         consumer,
-    //         producer,
-    //         data_store,
-    //     }
-    // }
-    pub async fn run(&self) {
-        println!("Receiver ON!");
-        let _ = &self.consumer.subscribe().await;
-        // for _n in 0..100 {
-        let mut total_count = 0;
-        let mut direct_sent_count = 0;
-        let mut db_insert_count = 0;
-        loop {
-            if let Ok(message) = &self.consumer.consume_message().await {
-                total_count += 1;
-                if headers_check(message.headers().unwrap()) {
-                    let new_message = &message;
-                    let headers = required_headers(new_message).expect("parsing headers failed");
-                    let message_deadline: DateTime<Utc> = DateTime::<Utc>::from_str(&headers[DEADLINE]).expect("String date parsing failed");
+    #[instrument(skip_all, fields(correlationId))]
```

Comment:
What does this `fields(correlationId)` do?

Reply:
ID to stitch the consumer of the delay scheduler to the delay process working. This field will be added as an attribute in the trace on which it can be filtered across different app traces

Reply:
When annotation processor will be handling this expression, where the value of `correlationId` will come from?